test/modern_bpf/event_class/event_class.cpp

```diff
@@ -185,7 +168,44 @@ void event_test::mark_all_64bit_syscalls_as_uninteresting()
 
 void event_test::enable_capture()
 {
-	pman_enable_capture();
+	bool tp_set[TP_VAL_MAX] = {0};
+	switch(m_event_type)
+	{
+	case PPME_PROCEXIT_1_E:
+		tp_set[SCHED_PROC_EXIT] = 1;
+		break;
+	case PPME_SCHEDSWITCH_6_E:
+		tp_set[SCHED_SWITCH] = 1;
+		break;
+	case PPME_SYSCALL_EXECVE_19_X:
+		// If event direction is BOTH, we are coming from a
+		// generic event_test::event_test(ppm_event_type event_type)
+		if (m_event_direction == BOTH_EVENT)
+		{
+			tp_set[SCHED_PROC_EXEC] = 1;
+			break;
+		}
+	case PPME_SYSCALL_CLONE_20_X:
+		// If event direction is BOTH, we are coming from a
+		// generic event_test::event_test(ppm_event_type event_type)
+		if (m_event_direction == BOTH_EVENT)
+		{
+			tp_set[SCHED_PROC_FORK] = 1;
+			break;
+		}
+	default:
+		if (m_event_direction & ENTER_EVENT)
+		{
+			tp_set[SYS_ENTER] = 1;
+		}
+		if (m_event_direction & EXIT_EVENT)
+		{
+			tp_set[SYS_EXIT] = 1;
+		}
+		break;
+	}
+	pman_enable_capture(tp_set);
+	clear_ring_buffers();
```

Comment:
This is needed to avoid `pman_enable_capture()`,internally calling some `bpf` syscalls, generated events to be fetched by the test loop.